Tribler/community/tunnel/tunnel_community.py

```diff
@@ -758,6 +799,28 @@ def _ours_on_created_extended(self, circuit, message):
             from Tribler.Core.simpledefs import NTFY_TUNNEL, NTFY_CREATED, NTFY_EXTENDED
             self.notifier.notify(NTFY_TUNNEL, NTFY_CREATED if len(circuit.hops) == 1 else NTFY_EXTENDED, circuit)
 
+
+    def on_introduction_request(self, messages):
+        exitnode = self.settings.become_exitnode
+        extra_payload = [exitnode]
+        super(TunnelCommunity, self).on_introduction_request(messages, extra_payload)
+        for message in messages:
+            pubkey = message.candidate.get_member().public_key
+            connectable = message.candidate.connection_type == u"public"
+            self.exit_candidates[pubkey] = message.payload.exitnode and connectable
```

Comment:
Then just remove the doctor and simply switch to yield verified peers, those have the connection type flag